packages/SwingSet/docs/vat-environment.md

```diff
@@ -0,0 +1,113 @@
+# The Vat Runtime JS Environment
+
+SwingSet JavaScript Vats are containers that run code in a confined and resource-limited environment, with orthogonal persistence and eventual-send-based access to external resources.
+
+The specific flavor of JavaScript provided by a vat is a frozen SES `Compartment`, with a few additions. Most JS environments (Node.js, web browsers) provide a combination of the baseline JavaScript [language globals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects) (`Object`, `Array`, etc) plus some number of "host objects". [Web browsers](https://developer.mozilla.org/en-US/docs/Web/API) offer things like `window`, `fetch`, `WebAssembly`, and `localStorage`, while [Node.js includes](https://nodejs.org/dist/latest-v14.x/docs/api/globals.html) ones like `require`, `process`. and `Buffer`. In general, all IO is provided by the host objects.
```

Comment:
(@tyg): "JavaScript flavor" 2) Why the "" around Compartment? If you want it to stand out as a new term, italics are the usual convention. 3) "(e.g. Node.js,..." 4) "(e.g. `Object`, `Array`..."

(@warner):  Are you recommending `The specific JavaScript flavor provided ..`? I wanted `specific` to bind to `flavor`. Or are you recommending `The JavaScript flavor provided ..`? I guess I wanted to introduce the notion that JS comes in different flavors, so "specific" seemed important.

(@warner):  Compartment: yeah, this is an introducing occurrence. It's also a code literal, and lives in the same category as `Object` and `Array` and `localStorage`. I guess I'll use "frozen SES `Compartment`" and let the fixed-width font do double-duty as a introducing-occurence marker.

Reply:
1) Fine then. I tend to try to change "x of y" to "y x" to both tighten and less passive, but you've got a reason, so...
2) Looks good to me.